flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java

```diff
@@ -423,8 +423,21 @@ private static Document getCachedIndexHtmlDocument(VaadinService service) {
     private static Document getIndexHtmlDocument(VaadinService service)
             throws IOException {
         DeploymentConfiguration config = service.getDeploymentConfiguration();
-        String index = FrontendUtils.getIndexHtmlContent(service);
-        if (index == null) {
+        String index;
+        try {
+            index = FrontendUtils.getIndexHtmlContent(service);
+        } catch (IOException e) {
+            if (config.getMode() == Mode.DEVELOPMENT_FRONTEND_LIVERELOAD) {
+                throw new IOException(
+                        "Unable to fetch index.html from the frontend development server, check the server logs that it is running");
+            } else {
+                throw new IOException(
+                        "Unable to find index.html. It should be available on the classpath");
+            }
+        }
+        if (index == null)
+
+        {
```

Comment:
```suggestion
        if (index == null) {
```
Formatting got a bit strange with multiple new lines.